### PR TITLE
add multicall2 - optionally allow call failures

### PR DESCRIPTION
**Pull request by JFrankfurt**
This is @WyseNynja's changes with a couple tweaks

- respects naming formats
- incorporates @moodysalem's req for `Multicall2.sol`

deployed on ROPSTEN RINKEBY GÖRLI KOVAN here:
`0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696`
https://kovan.etherscan.io/address/0x5ba1e12693dc8f9c48aad8770482f4739beed696

### CODE REVIEW

File: src/Multicall2.sol [hunk 1]
```diff
@@ -0,0 +1,72 @@
+pragma solidity >=0.5.0;
+pragma experimental ABIEncoderV2;
+
+/// @title Multicall2 - Aggregate results from multiple read-only function calls
+/// @author Michael Elliot <mike@makerdao.com>
+/// @author Joshua Levine <joshua@makerdao.com>
+/// @author Nick Johnson <arachnid@notdot.net>
+
+contract Multicall {
```
Comment (WyseNynja) [inline src/Multicall2.sol:9]: This should probably be `contract Multicall2 {`

**Review by WyseNynja — COMMENTED** [review]
Thanks!

**Comment by JFrankfurt** [comment]
@michaelelliot You have any thoughts on this?

**Comment by banteg** [comment]
Dev alpha leak: You can use Multicall2 _today_ on any network and even to query historical data.

Use Geth [state override feature](https://geth.ethereum.org/docs/rpc/ns-eth#3-object---state-override-set) to set the bytecode of any contract to the bytecode of deployed Multicall2 prior to `eth_call`.

**Comment by NoahZinsmeister** [comment]
also deployed on mainnet now at the same address

https://etherscan.io/address/0x5ba1e12693dc8f9c48aad8770482f4739beed696

**Pull request merged**